[addons] rename kodi-addon-dev-kit folder to kodi-dev-kit and clang cleanup of addon GUI sources

AlwinEsch opened the pull request:
## Description

Since long time was wanted to rename them to a smaller name and remove "addon" from it as it can also maybe become other parts.

As shorter folder name is in all cases better as before :-D

### Commit 2:
-------------
**[addons][gui] whole clang cleanup of all gui related sources**

This makes the clang cleanup of all GUI source files related to addons where was previous moved to own folder.

There was some parts inside where previous has not matched the wanted clang format of Kodi.

Also are the doxygen doc parts field a bit cleaned up to prevent long lines.

> About second commit here, was gone down on begin by https://github.com/xbmc/xbmc/pull/18317 and added here, as the changes on this request also not affect the code process work (only code cleanup).

## User related

No user relationship

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons] rename kodi-addon-dev-kit folder to kodi-dev-kit

Since long time was wanted to rename them to a smaller name and
remove "addon" from it as it can also maybe become other parts.

As shorter folder name is in all cases better as nefore :-D

garbear reviewed (approved):
I'm surprised by how many files this touches, but changes LGTM

AlwinEsch committed:
[addons][gui] whole clang cleanup of all gui related sources

This makes the clang cleanup of all GUI source files related to
addons where was previous moved to own folder.

There was some parts inside where previous has not matched the
wanted clang format of Kodi.

Also are the doxygen doc parts field a bit cleaned up to prevent
long lines.
